hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -261,6 +261,15 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new RuntimeException("Interrupted in call to rename", e);
+    } catch (RuntimeException e) {
+      // in case of table already exists,
+      // Hive rename operation throws exception as
+      // java.lang.RuntimeException:InvalidOperationException(message:new table <> already exists)
+      if (e.getMessage().contains(String.format("new table %s already exists)", to))) {
```

Comment:
I think here we should examine the exact cause rather than just looking at the exception message. We don't want to throw an `AlreadyExistsException` when the error is something else